controllers/platformnetwork_controller.go

```diff
@@ -661,38 +742,291 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
+		_ = r.CloudManager.StartMonitor(m, msg)
+	}
+	return common.NewResourceConfigurationDependency("waiting for system to be in locked state before reconfiguring network")
+}
+
+// ReconcileResource interacts with the system API in order to reconcile the
+// state of a data network with the state stored in the k8s database.
+func (r *PlatformNetworkReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork, request_namespace string) (err error) {
+	var inSync bool
+	oldStatus := instance.Status.DeepCopy()
+
+	if instance.DeletionTimestamp.IsZero() {
+		update_pn, err := r.IsPlatformNetworkInSync(client, instance)
+		if err != nil {
+			return err
+		}
+		if instance.Status.DeploymentScope == cloudManager.ScopeBootstrap {
+			inSync = !update_pn
+			if instance.Spec.Type == cloudManager.MgmtNetworkType ||
+				instance.Spec.Type == cloudManager.OAMNetworkType ||
+				instance.Spec.Type == cloudManager.AdminNetworkType {
+				// Just check current config vs profile for networks of type mgmt, oam and admin
+				// Update instance.Status.InSync to false if there is a difference and vice-versa.
+				// instance.Status.Reconciled will always be set to true to avoid raising alarms in Day-1
+				// for above mentioned network types.
+				if instance.Status.InSync != inSync {
+					r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "synchronization has changed to: %t", inSync)
+				}
+				instance.Status.InSync = inSync
+				instance.Status.Reconciled = true
+				err2 := r.Client.Status().Update(context.TODO(), instance)
+				if err2 != nil {
+					logPlatformNetwork.Error(err2, "failed to update platform network status")
+				}
+				return err2
+			} else {
+				// Try to reconcile rest of the networks and update the status accordingly.
+				_, err = r.ReconcileAddressPoolAndNetwork(client, instance)
+				if err != nil {
+					return err
+				}
 			}
+		} else if instance.Status.DeploymentScope == cloudManager.ScopePrincipal {
+			if !(instance.Status.InSync && instance.Status.Reconciled) {
+				host, is_aiosx, err := r.AIOSXHost(client)
+				if err != nil {
+					return err
+				}
+				if is_aiosx {
+					logPlatformNetwork.Info("Proceeding with network reconfiguration on AIO-SX")
+					if instance.Spec.Type == cloudManager.MgmtNetworkType {
+
+						if host.AdministrativeState == hosts.AdminLocked && host.OperationalStatus == hosts.OperDisabled {
+
+							if update_pn {
+								_, err = r.ReconcileAddressPoolAndNetwork(client, instance)
+								if err != nil {
+									return err
+								}
+							}
+
+							// Trigger network reconciliation on host controller by creating annotation
+							// on host instance if the network type is management.
+							host_instance, err := r.GetHostInstance(host.Hostname, request_namespace)
+							if err != nil {
+								logPlatformNetwork.Error(err, "Failed to get resource for host namespace %v", request_namespace)
```

Comment:
Failed to get host resource from namespace, similiar comments apply in the rest logs

Reply:
> Failed to get host resource from namespace, similiar comments apply in the rest logs

Done.